language/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -345,24 +345,32 @@ impl<'env> SpecTranslator<'env> {
 // ============================
 
 impl<'env> SpecTranslator<'env> {
-    pub(crate) fn finalize(&self) {
+    pub(crate) fn finalize(&mut self) {
         self.translate_choice_functions();
     }
 
     /// Translate lifted functions for choice expressions.
-    fn translate_choice_functions(&self) {
+    fn translate_choice_functions(&mut self) {
```

Comment:
You don't need the &mut here. See e.g. line 215 how to fork a new spec translator with a specialized type instantiation.